### PR TITLE
feat(deps): move from leiningen to boot, remove dev folder, remove schema

### PR DESCRIPTION
**Pull request by thenonameguy**
Cloverage feature still missing, I think I'll write `boot-cloverage` for the community, based on `lein-cloverage`, it seems pretty easy.

### CODE REVIEW

File: build.boot [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+(require
+ '[github-changelog.core :refer [changelog]]
+ '[adzerk.boot-test :as boot-test]
+ '[tolitius.boot-check :as check]
+ '[adzerk.bootlaces :refer :all])
+
+(bootlaces! +version+)
+
+(task-options!
+ pom {:project 'hu.ssh/github-changelog
```
Comment (raszi) [inline build.boot]: This is not the project name anymore.

**Comment by thenonameguy** [comment]
Turns out the JUnit output is not as handy for doing coverages as I thought. Let's discuss tomorrow how we will do metrics on our code.

File: src/github_changelog/core.clj [hunk 2]
```diff
@@ -1,63 +1,48 @@
...
-        tags (git/tags git)]
-    (map (partial assoc-commits git) (parse-tags tags))))
-
-(s/defn find-pull :- (s/maybe Pull)
-  [pulls :- [Pull]
-   sha :- Sha]
+(defn assoc-commits [git {:keys [from sha] :as tag}]
+  (assoc tag :commits (git/commits git from sha)))
+
+
```
Comment (thenonameguy) [inline src/github_changelog/core.clj]: Double newline.

File: test/github_changelog/conventional_test.clj [hunk 3]
```diff
@@ -1,43 +1,39 @@
...
-    (let [pull (create-pull {:body "Fixes #1" :base {:repo {:html_url repo-url}}})]
+    (let [pull (complete-pull {:body "Fixes #1" :base {:repo {:html_url repo-url}}})]
       (is (= [["#1" (str repo-url "/issues/1")]] (conventional/parse-issues config pull))))))
 
+(conventional/parse-pull config (complete-pull {:title "this is just a PR"}))
+
 (deftest parse-pull
   (testing "with a correct formats"
-    (are [title] (s/validate Change (conventional/parse-pull config (create-pull {:title title})))
+    (are [title] (not= nil (conventional/parse-pull config (complete-pull {:title title})))
```
Comment (thenonameguy) [inline test/github_changelog/conventional_test.clj]: I could'nt come up with a better assertion on the spot, than this, we can possibly do better.

File: test/github_changelog/schema_generators.clj [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+(defn complete
+  ([gen] #(complete % gen))
+  ([partial-datum gen]
+   (-> (gen/generate gen)
+       (merge partial-datum))))
+
+(defmacro defgen [fn-name & body]
+  `(do
+     (def ~fn-name (gen/hash-map ~@body))
+     (def ~(symbol (str "complete-" fn-name)) (complete ~fn-name))))
```
Comment (thenonameguy) [inline test/github_changelog/schema_generators.clj]: The rockstar macro of this PR.

File: test/github_changelog/util_test.clj [hunk 5]
```diff
@@ -7,5 +7,5 @@
   (is (= "foobar" (util/strip-trailing "foobar" "!!"))))
 
 (deftest extract-params
-  (is (= {}) (util/extract-params ""))
-  (is (= {:a 1 :b 2}) (util/extract-params "a=1&b=2")))
+  (is (= {} (util/extract-params "")))
+  (is (= {:a "1" :b "2"} (util/extract-params "a=1&b=2"))))
```
Comment (thenonameguy) [inline test/github_changelog/util_test.clj]: I don't even know how these ran before, wtf?

File: src/github_changelog/conventional.clj [hunk 6]
```diff
@@ -1,58 +1,46 @@
...
-  [pattern :- s/Str]
-  (let [close-keywords ["close" "closes" "closed" "fix" "fixes" "fixed" "resolve" "resolves" "resolved"]]
-    (re-pattern (format "(?i:%s) %s" (string/join \| close-keywords) pattern))))
+(def close-keywords ["close" "closes" "closed" "fix" "fixes" "fixed" "resolve" "resolves" "resolved"])
+
+(defn fixes-pattern
+  ([pattern] (fixes-pattern pattern close-keywords))
+  ([pattern closing-words]
+   (re-pattern
+    (format "(?i:%s) %s"
```
Comment (raszi) [inline src/github_changelog/conventional.clj]: Wrong indenting.

File: test/github_changelog/conventional_test.clj [hunk 7]
```diff
@@ -1,43 +1,39 @@
...
-    [github-changelog.conventional :as conventional]
-    [github-changelog.schema :refer [Config Pull Change]]
-    [github-changelog.schema-generators :refer [generators]]
-    [github-changelog.schema-complete :refer [complete]]
-    [clojure.test :refer :all]
-    [clojure.string :refer [join]]
-    [schema.core :as s]))
+   [github-changelog
+    [conventional :as conventional]
+    [schema-generators :as sgen :refer [complete-pull complete-config]]]
```
Comment (raszi) [inline test/github_changelog/conventional_test.clj]: Please remove `:as sgen`

File: src/github_changelog/conventional.clj (hunk 6 above)
Comment (thenonameguy) [inline src/github_changelog/conventional.clj]: This is how my emacs wants it :(

**Pull request merged**